jOOL-java-8/src/main/java/org/jooq/lambda/Agg.java

```diff
@@ -1163,4 +1163,180 @@ else if (l.size() == 1)
             s -> s.map(Objects::toString).orElse("")
         );
     }
+
+    /**
+     * Realize of linear regression functions.
+     */
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrCountLong()</code> function.
+     */
+    public static  Collector<Tuple2<Double,Double>, ?, Optional<Long>> regrCountLong() {
+        return regrCountLong(t -> t.v1, t -> t.v2);
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrCountLong()</code> function.
+     */
+    public static <T> Collector<T, ?, Optional<Long>> regrCountLong(ToDoubleFunction<? super T> functionX, ToDoubleFunction<? super T> functionY) {
+        return collectingAndThen(toList(), l -> {
+            if (l.isEmpty())
+                return Optional.empty();
+            long result = 0;
+            for (T t : l)
+                result += t==null ? 0:1;
+            return Optional.of(result);
+        });
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrAvgXDouble()</code> function.
+     */
+    public static  Collector<Tuple2<Double,Double>, ?, Optional<Double>> regrAvgXDouble() {
+        return regrAvgXDouble(t -> t.v1, t -> t.v2);
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrAvgXDouble()</code> function.
+     */
+    public static <T> Collector<T, ?, Optional<Double>> regrAvgXDouble(ToDoubleFunction<? super T> functionX, ToDoubleFunction<? super T> functionY) {
+        return collectingAndThen(toList(), l -> {
+            if (l.isEmpty())
+                return Optional.empty();
+            else if (l.size() == 1)
+                return Optional.of(0.0);
+            return Optional.of(avg0(l,functionX));
+        });
+    }
```

Comment:
Again, this isn't really much different from the ordinary `Collectors.averagingDouble()` collector, so why do we need it? Just because SQL has it? I mean, we could offer it, but then it should just delegate to the relevant JDK `Collectors.averagingDouble()` call. But I doubt it's useful.